extensions/amp-access/0.1/amp-access.js

```diff
@@ -770,17 +777,43 @@ export class AccessService {
   }
 
   /**
+   * Runs the login flow using one of the predefined urls in the amp-access config
+   *
+   * @param {string} type Type of login defined in the config
+   * @return {!Promise}
+   */
+  loginWithType_(type) {
+    // Login URL should always be available at this time.
+    const loginUrl = user().assert(this.loginUrlMap_[type],
+        'Login URL is not ready: %s', type);
+    user().assert(this.loginConfig_[type],
+        'Login URL is not configured: %s', type);
+    return this.login_(loginUrl, type);
+  }
+
+  /**
+   * Runs the login flow opening the given url in the login window.
+   *
+   * @param {string} url
+   * @return {!Promise}
+   */
+  loginWithUrl(url) {
+    return this.login_(url, url);
```

Comment:
Ah, we do have different purchase types, pay per use or single item sale (time passes), which we essentially use as the different login urls, so that would probably be the correct thing to pass along here.